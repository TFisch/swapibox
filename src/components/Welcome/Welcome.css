.star-wars {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic one', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify; }

.crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear; }

.crawl > .title {
  font-size: 90%;
  text-align: center; }

.crawl > .title {
  font-size: 90%;
  text-align: center; }

@keyframes crawl {
  0% {
    top: 0;
    transform: rotate(0deg) translateZ(0); }
  100% {
    top: -6000px;
    transform: rotateX 0deg translateZ(-2500px); } }

.fade {
  position: relative;
  width: 100%;
  min-height: 60vh;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1; }
