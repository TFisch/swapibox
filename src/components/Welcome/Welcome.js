import React from 'react';
import PropTypes from 'prop-types';

 const Welcome = ( { crawlingText }) => {
  const { episode, title, scrawl } = crawlingText
  return (
    <section className="star-wars">
      <div className="scrawl">
        <div className="title">
          <p>{episode}</p>
          <h1>{title}</h1>
        </div>
        <p>{scrawl}</p>     
      </div>
    </section>
  )
}

Welcome.propTypes = {
  crawlingText: PropTypes.object
};

export default Welcome;