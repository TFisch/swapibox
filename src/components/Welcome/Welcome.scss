.star-wars {
  display: flex;
  justify-content: center;
  height: 800px;
  perspective: 400px;
  color: #feda4a;
  font-family: 'Pathway Gothic one', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  text-align: justify;
}

.crawl {
  position: relative;
  top: -100px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotate(20deg) translateZ(0);
  }
  100% {
    top: -6000px;
    transform: rotateX (25deg) translateZ(-2500px);
  }
}