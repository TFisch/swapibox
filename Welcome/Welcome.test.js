import React from 'react';
import { shallow } from 'enzyme';
import Welcome from './Welcome.js';

describe('WELCOME', () => {
  it('Should match snapshot', () =>{
    const wrapper = shallow(<Welcome crawlingText={{}} />);

    expect(wrapper.html()).toMatchSnapshot();
  });
});

