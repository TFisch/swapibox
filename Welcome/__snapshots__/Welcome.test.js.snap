// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WELCOME Should match snapshot 1`] = `"<div class=\\"fade\\"><section class=\\"star-wars\\"><div class=\\"crawl\\"><div class=\\"title\\"><p></p><h1></h1></div><p></p></div></section></div>"`;
